pkg/controller/controller_test.go

```diff
@@ -0,0 +1,48 @@
+package controller
+
+import (
+	"context"
+	"testing"
+	"time"
+)
+
+func TestController(t *testing.T) {
+	ctx, cncl := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cncl()
+	ctx1, cncl1 := context.WithCancel(context.Background())
+	defer cncl1()
+
+	type args struct {
+		ctx   context.Context
+		flags map[string]string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "context deadline",
+			args: args{
+				ctx:   ctx,
+				flags: map[string]string{"lang": "go", "since": "weekly"},
+			},
+			wantErr: true,
+		},
+		{
+			name: "pass",
```

Comment:
Test name should be self-explanatory. some thing like `Success - User win the game`

Reply:
You can create function to separate out the user input and the business logic. So that you can right the better unit test functions.

Reply:
for user input I have created IsCorrectGuess function in utils package with checking the tolerance with the input values.